packages/vite-plugin-nitro/src/lib/build-sitemap.ts

```diff
@@ -0,0 +1,73 @@
+import { SitemapConfig } from './options';
+import * as fs from 'fs';
+import { XMLBuilder } from 'xmlbuilder2/lib/interfaces';
+import { create } from 'xmlbuilder2';
+import { UserConfig } from 'vite';
+import * as path from 'path';
+
+export type PagesJson = {
+  page: string;
+  lastMod: string;
+};
+
+export async function buildSitemap(
+  config: UserConfig,
+  sitemapConfig: SitemapConfig,
+  routes: string[] | (() => Promise<(string | undefined)[]>)
+) {
+  const routeList: string[] = await optionHasRoutes(routes);
+
+  if (routeList.length) {
+    const sitemapData: PagesJson[] = routeList.map((page: string) => ({
+      page: `https://www.${sitemapConfig.domain}${page.replace(/^\/+/g, '')}`,
+      lastMod: new Date().toISOString().split('T')[0],
+    }));
+
+    const sitemap = createXml('urlset');
+
+    for (const item of sitemapData) {
+      const page = sitemap.ele('url');
+      page.ele('loc').txt(item.page);
+      page.ele('lastmod').txt(item.lastMod);
+    }
+
+    const mapPath = `${path.resolve(
+      'dist',
+      config.root!,
```

Comment:
Let's add a fallback root

```suggestion
      config.root || '.',
```